dask/annotation.py

```diff
@@ -0,0 +1,19 @@
+import functools
+
+
+def annotate_func(func: callable, annotation: dict) -> callable:
+    ret = functools.partial(func)
+    if not hasattr(ret, "_dask_annotation_dict"):
+        ret._dask_annotation_dict = {}
+        functools.update_wrapper(
+            ret, func, functools.WRAPPER_ASSIGNMENTS + ("__repr__",)
+        )
+    ret._dask_annotation_dict.update(annotation)
+    return ret
```

Comment:
I'm gussing that `functools.partial` is used here to avoid the case where we can't attach attributes to some functions?

Reply:
Yes, and also to avoid annotating unintentionally. E.g., setting a new attribute directly on `getitem` will annotate all use `getitem`.

Reply:
Ah right.  That makes sense.  Thank you for the explanation.